<?xml version="1.0"?>

<launch>

  <param name="use_sim_time" type="bool" value="True"/>

  <!-- SLAM (robot side) -->
  <!-- args: "delete_db_on_start" and "udebug" -->
  <group ns="rtabmap">
    <node name="rtabmap" pkg="rtabmap_slam" type="rtabmap" output="screen" args="--delete_db_on_start">
	  <param name="frame_id" type="string" value="base_footprint"/>

	  <param name="subscribe_depth" type="bool" value="true"/>
	  <param name="subscribe_laserScan" type="bool" value="true"/>

	  <remap from="odom" to="/base_controller/odom"/>
	  <remap from="scan" to="/base_scan"/>

	  <remap from="rgb/image" to="/camera/data_throttled_image"/>
  	  <remap from="depth/image" to="/camera/data_throttled_image_depth"/>
  	  <remap from="rgb/camera_info" to="/camera/data_throttled_camera_info"/>

  	  <param name="rgb/image_transport" type="string" value="compressed"/>
     	  <param name="depth/image_transport" type="string" value="compressedDepth"/>

	  <param name="queue_size" type="int" value="10"/>

	  <!-- RTAB-Map's parameters: do "rosrun rtabmap rtabmap (double-dash)params" to see the list of available parameters. -->
	  <param name="RGBD/ScanMatchingSize" type="string" value="1"/>             <!-- Do odometry correction with consecutive laser scans -->
	  <param name="RGBD/LocalLoopDetectionSpace" type="string" value="true"/>   <!-- Local loop closure detection (using estimated position) with locations in WM -->
	  <param name="RGBD/LocalLoopDetectionTime" type="string" value="false"/>    <!-- Local loop closure detection with locations in STM -->
	  <param name="Mem/BadSignaturesIgnored" type="string" value="false"/>      <!-- Don't ignore bad images for 3D node creation (e.g. white walls) -->
	  <param name="LccIcp/Type" type="string" value="2"/>                       <!-- Loop closure transformation refining with ICP: 0=No ICP, 1=ICP 3D, 2=ICP 2D -->
	  <param name="LccIcp2/Iterations" type="string" value="100"/>
	  <param name="LccIcp2/VoxelSize" type="string" value="0"/>
	  <param name="LccBow/MinInliers" type="string" value="5"/>                 <!-- 3D visual words minimum inliers to accept loop closure -->
	  <param name="LccBow/MaxDepth" type="string" value="4.0"/>                 <!-- 3D visual words maximum depth 0=infinity -->
	  <param name="LccBow/InlierDistance" type="string" value="0.1"/>           <!-- 3D visual words correspondence distance -->
	  <param name="RGBD/AngularUpdate" type="string" value="0.01"/>             <!-- Update map only if the robot is moving -->
	  <param name="RGBD/LinearUpdate" type="string" value="0.01"/>              <!-- Update map only if the robot is moving -->
	  <param name="Rtabmap/TimeThr" type="string" value="700"/>
	  <param name="Mem/RehearsalSimilarity" type="string" value="0.45"/>
	  <param name="Mem/RehearsedNodesKept" type="string" value="false"/>
    </node>
  </group>

  <!-- send AZIMUT 3 urdf to param server -->
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find az3_description)/robots/azimut_3_laser.urdf.xacro'" />

  <!-- Visualisation -->
  <include file="$(find rtabmap_legacy)/launch/azimut3/az3_mapping_client.launch"/>

</launch>
